lib/Command/Room/Update.php

```diff
@@ -0,0 +1,137 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Command\Room;
+
+use Exception;
+use OC\Core\Command\Base;
+use OCA\Talk\Exceptions\RoomNotFoundException;
+use OCA\Talk\Manager;
+use OCP\IConfig;
+use OCP\IUserManager;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class Update extends Base {
+	use TRoomCommand;
+
+	/** @var IConfig */
+	private $config;
+
+	/** @var IUserManager */
+	public $userManager;
+
+	/** @var Manager */
+	public $manager;
+
+	public function __construct(IConfig $config, IUserManager $userManager, Manager $manager) {
+		parent::__construct();
+
+		$this->config = $config;
+		$this->userManager = $userManager;
+		$this->manager = $manager;
+	}
+
+	protected function configure(): void {
+		$this
+			->setName('talk:room:update')
+			->setDescription('Updates a room')
+			->addArgument(
+				'token',
+				InputArgument::REQUIRED,
+				'The token of the room to update'
+			)->addOption(
+				'name',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Sets a new name for the room'
+			)->addOption(
+				'public',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Modifies the room to be a public room (value 1) or private room (value 0)'
+			)->addOption(
+				'readonly',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Modifies the room to be a room with read-only (value 1) or read-write (value 0) access'
+			)->addOption(
+				'password',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Sets a new password for the room; pass a empty value to remove password protection'
+			);
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): ?int {
+		$token = $input->getArgument('token');
+		$name = $input->getOption('name');
+		$public = $input->getOption('public');
+		$readOnly = $input->getOption('readonly');
+		$password = $input->getOption('password');
+
+		if (!in_array($public, [null, '0', '1'], true)) {
+			$output->writeln('<error>Invalid value for option "--public" given.</error>');
+			return 1;
+		}
+
+		if (!in_array($readOnly, [null, '0', '1'], true)) {
+			$output->writeln('<error>Invalid value for option "--readonly" given.</error>');
+			return 1;
+		}
+
+		try {
+			$room = $this->manager->getRoomByToken($token);
+		} catch (RoomNotFoundException $e) {
+			$output->writeln('<error>Room not found.</error>');
+			return 1;
+		}
+
+		try {
+			if ($name !== null) {
+				$this->setRoomName($room, $name);
```

Comment:
This works by accident for one-to-one conversations at the moment.

I would say that we should check the type after each `getRoomByToken` explicitly